docs/source/developer_guide/dependencies.rst

```diff
@@ -58,6 +64,43 @@ Estimators with a soft dependency need to ensure the following:
    ``run_test_for_class`` usage to decorate a test. See ``utils.tests.test_plotting``
    for an example of ``_check_soft_dependencies`` usage.
 
+Avoiding module-level imports of soft dependencies
+-------------------------------------
```

Comment:
number of underlines should be the same as number of letter characters, and level should be the same as previous section